kolibri/core/content/utils/content_request.py

```diff
@@ -352,24 +435,53 @@ def process_metadata_import(incomplete_downloads_without_metadata):
     :param incomplete_downloads_without_metadata: a ContentDownloadRequest queryset
     :type incomplete_downloads_without_metadata: django.db.models.QuerySet
     """
-    peer = _get_preferred_network_location(version_filter=">=0.16.0")
-    if not peer:
-        # can't import metadata without peers having minimum version 0.16.0
-        logger.info("No acceptable peer network device for importing content metadata")
-        return
+    source_instance_ids = list(
+        incomplete_downloads_without_metadata.values_list(
+            "source_instance_id", flat=True
+        ).distinct()
+    )
+    # append `None` which will bypass the explicit peer selection and use a trusted peer
+    # of a recent sync, if available, during the last step of the process
+    source_instance_ids.append(None)
+    peers = []
+
+    for source_instance_id in source_instance_ids:
+        # get the peer for this source instance
+        peer = _get_preferred_network_location(
```

Comment:
I wonder if the _get_preferred_network_location function could be implemented slightly differently to accept the full list of source_instance_ids and return a generator that can then be iterated below `for peer in peers` and we just keep going through the list of peers generated by that until all the requested resources have been imported.

This could also then include the 'trusted' peer behaviour even in the case where source_instance_ids have been passed too.

Reply:
I think that's a good idea and would make it more flexible for other usage. I'll implement that